osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -729,22 +733,28 @@ private void updateYPositions()
         {
             visibleItems.Clear();
 
-            float currentY = visibleHalfHeight;
+            // visibleHalfHeight includes BleedTop, but Scroll screenspace doesn't, so we remove it here
+            float currentY = visibleHalfHeight - BleedTop;
 
             scrollTarget = null;
 
+            double elapsed = Clock.CurrentTime - timeSinceLastYUpdate;
+
             foreach (CarouselItem item in root.Children)
             {
-                if (item.Filtered.Value)
-                    continue;
-
                 switch (item)
                 {
                     case CarouselBeatmapSet set:
```

Comment:
Since you've moved the `item.Filtered` early exit inside the `switch`, i'd propose changing this to a direct cast or typed-`if` to avoid future slip-ups if other `case`s were added:

```csharp
if (!(item is CarouselBeatmapSet set)) continue;
```

or, if that isn't required:

```csharp
CarouselBetamapSet set = (CarouselBeatmapSet)item;
```

Reply:
Right now carousel root only holds beatmap sets so checking this isn't required. I would change the foreach to happen over beatmapSets, avoided that because it would unnest this entire switch causing unnecessary whitespace changes in the diff view.

Reply:
Whitespace changes are the least of our concerns ever.